src/main/scala/gitbucket/core/controller/IndexController.scala

```diff
@@ -55,7 +55,7 @@ trait IndexControllerBase extends ControllerBase {
   post("/signin", signinForm){ form =>
     authenticate(context.settings, form.userName, form.password) match {
       case Some(account) => signin(account)
-      case None          => redirect("/signin")
+      case None          => gitbucket.core.html.signin(form.userName, form.password)
```

Comment:
In this case, we should use flash attribute. Reference:
https://github.com/gitbucket/gitbucket/blob/7c484297d7f0343c78b64c11dc668de613cc9cb7/src/main/scala/gitbucket/core/controller/SystemSettingsController.scala#L150-L168